packages/router/src/ctx/approver.ts

```diff
@@ -1,13 +1,25 @@
 import { sendPopupRequest, spawnDetachedPopup } from '@penumbra-zone/types/src/internal-msg/popup';
 import { TxApproval } from '@penumbra-zone/types/src/internal-msg/tx-approval';
 import { AuthorizeRequest } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/custody/v1alpha1/custody_pb';
+import { Jsonified } from '@penumbra-zone/types';
+import { DenomMetadata } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/asset/v1alpha1/asset_pb';
 
-export const getTxApproval = async (req: AuthorizeRequest): Promise<void> => {
+export const getTxApproval = async (
+  req: AuthorizeRequest,
+  denomMetadataByAssetId: Record<string, Jsonified<DenomMetadata>>,
+): Promise<void> => {
   await spawnDetachedPopup('popup.html#/approval/tx');
 
+  /**
+   * @todo: Should this include a request ID so as not to cross wires with other
+   * requests?
+   */
```

Comment:
yes - i think we should implement this in this PR. just use crypto.getrandomuuid to make one at the moment the request is sent, and check that the response contains the same uuid.

Reply:
in the zustand slice, we should
 
1) never return the uuid as part of a selector
2) throw if we are assigning data but a uuid is already present
3) clear the uuid when resolving the approval